graphql/resolvers/directives.js

```diff
@@ -0,0 +1,73 @@
+import fs from 'fs';
+import path from 'path';
+import { forEachField } from 'graphql-tools';
+import { getArgumentValues } from 'graphql/execution/values';
+import jwt from 'jsonwebtoken';
+import debug from 'debug';
+
+import { AuthorizationError } from '../errors';
+
+const log = debug('fcc:resolvers:directives');
+const certPath = process.cwd() + '/public.pem';
+const cert = fs.readFileSync(certPath);
+
+function isAuth(ctx) {
+  const token = ctx.headers.authorization;
+  if (!token) {
+    throw new AuthorizationError({
+      message: 'You must supply a JSON Web Token for authorization!'
+    });
+  }
+  try {
+    return jwt.verify(token.replace('Bearer ', ''), cert);
+  } catch (err) {
```

Comment:
It would be a courtesy to our users and future selves to tell them if their token has expired or something else went wrong. Glancing at [JsonWebTokenError](https://github.com/auth0/node-jsonwebtoken#jsonwebtokenerror) it could be fine to bubble those up to the user?

Reply:
Sounds reasonable